paddle/fluid/memory/allocation/allocator_facade.cc

```diff
@@ -252,12 +258,40 @@ class AllocatorFacadePrivate {
 #endif
   }
 
-  void WrapZeroSizeAllocator() {
-    for (auto& pair : allocators_) {
-      pair.second =
-          std::make_shared<ZeroSizeAllocator>(pair.second, pair.first);
+  class ZeroSizeAllocator : public Allocator {
```

Comment:
Why do you move `ZeroSizeAllocator` here?

Reply:
Because I think that it may be too complex that each allocator must be decorated by `ZeroSizeAllocator`. `ZeroSizeAllocator` should only handle allocation request size = 0 alone.